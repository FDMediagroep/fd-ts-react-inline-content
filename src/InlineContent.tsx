import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
