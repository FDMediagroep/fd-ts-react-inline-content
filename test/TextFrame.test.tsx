import React from 'react';
import TestRenderer from 'react-test-renderer';
import { TextFrame } from '../src/InlineContent';

describe('TextFrame', ()  => {
    const title = "140%";
    const paragraph = "Curabitur quis luctus est, vitae sodales enim";
    const image = {
        alt: 'Het zwarte schaap: Anglo Irish',
        src: 'https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1280'
    };
    const sourceSets = [{
        media: '(max-width: 640px)',
        srcSet: 'https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=600 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1200 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1800 3x'
    }, {
        media: '(max-width: 860px)',
        srcSet: 'https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=361 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=722 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1083 3x'
    }];

    it('should render align right correctly', () => {
        let component = TestRenderer.create(<TextFrame {...{title, paragraph}} />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<TextFrame {...{title, paragraph}} align='right'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render align block correctly', () => {
        let component = TestRenderer.create(<TextFrame {...{title, paragraph}} align='block'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render with image align right correctly', () => {
        let component = TestRenderer.create(<TextFrame {...{title, paragraph, image, sourceSets}} />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<TextFrame {...{title, paragraph, image, sourceSets}} align='right'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render with image align block correctly', () => {
        let component = TestRenderer.create(<TextFrame {...{title, paragraph, image, sourceSets}} align='block'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
