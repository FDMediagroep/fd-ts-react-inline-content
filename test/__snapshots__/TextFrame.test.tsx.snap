// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFrame should render align block correctly 1`] = `
<div
  className="inline-content text-frame block"
>
  <h3
    className="outline"
  >
    140%
  </h3>
</div>
`;

exports[`TextFrame should render align right correctly 1`] = `
<div
  className="inline-content text-frame right"
>
  <h3
    className="outline"
  >
    140%
  </h3>
</div>
`;

exports[`TextFrame should render align right correctly 2`] = `
<div
  className="inline-content text-frame right"
>
  <h3
    className="outline"
  >
    140%
  </h3>
</div>
`;

exports[`TextFrame should render with image align block correctly 1`] = `
<div
  className="inline-content text-frame block"
>
  <h3
    className="outline"
  >
    140%
  </h3>
  <figure
    className="fd-teaser-figure"
  >
    <picture>
      <source
        media="(max-width: 640px)"
        srcSet="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=600 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1200 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1800 3x"
      />
      <source
        media="(max-width: 860px)"
        srcSet="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=361 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=722 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1083 3x"
      />
      <img
        alt="Het zwarte schaap: Anglo Irish"
        src="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1280"
      />
    </picture>
  </figure>
</div>
`;

exports[`TextFrame should render with image align right correctly 1`] = `
<div
  className="inline-content text-frame right"
>
  <h3
    className="outline"
  >
    140%
  </h3>
  <figure
    className="fd-teaser-figure"
  >
    <picture>
      <source
        media="(max-width: 640px)"
        srcSet="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=600 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1200 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1800 3x"
      />
      <source
        media="(max-width: 860px)"
        srcSet="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=361 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=722 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1083 3x"
      />
      <img
        alt="Het zwarte schaap: Anglo Irish"
        src="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1280"
      />
    </picture>
  </figure>
</div>
`;

exports[`TextFrame should render with image align right correctly 2`] = `
<div
  className="inline-content text-frame right"
>
  <h3
    className="outline"
  >
    140%
  </h3>
  <figure
    className="fd-teaser-figure"
  >
    <picture>
      <source
        media="(max-width: 640px)"
        srcSet="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=600 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1200 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1800 3x"
      />
      <source
        media="(max-width: 860px)"
        srcSet="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=361 1x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=722 2x, https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1083 3x"
      />
      <img
        alt="Het zwarte schaap: Anglo Irish"
        src="https://fd-external-development.imgix.net/c4489dd02274747af8e77d1451eb1ca4b458e57e.jpg?auto=format&fm=jpg&fit=max&q=45&w=1280"
      />
    </picture>
  </figure>
</div>
`;
